### PR TITLE
[CUDA] Update GroupNorm and Add SkipGroupNorm

### PR DESCRIPTION
**Pull request by tianleiwu**
### Description

* Update GroupNorm kernel to support number of channels used in SD XL refiner.
* Add epsilon in kernel
* Add parity and performance test script
* Add a new operator SkipGroupNorm to support skip and bias inputs.
* Remove many limitations including max batch size, max number of groups, c % cPerBlock ==0 etc.

### Motivation and Context

Update GroupNorm to support SD XL Refiner and beyond.

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu [hunk 1]
```diff
@@ -397,39 +585,38 @@
...
-  if (batch_size > static_cast<int>(kMaxGroupNormBatchSize)) {
-    return ORT_MAKE_STATUS(ONNXRUNTIME, StatusCode::NOT_IMPLEMENTED,
-                           "only support batch_size <= 32. Got", batch_size);
-  }
+    bool use_silu,
+    bool broadcast_skip,
+    int channels_per_block) {
+  GroupNormNHWCParams<T> params;
+
+  int32_t cPerGroup = num_channels / num_groups;
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:593]: >cPerGroup [](http://example.com/codeflow?start=10&length=9)

nit: follow the naming convention

File: onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu [hunk 2]
```diff
@@ -128,21 +186,94 @@
...
   int32_t ni = blockIdx.z;
-  // The channel loaded by that thread (2 channels per thread for F16x2).
-  int32_t ci = blockIdx.x * params.cPerBlock + threadIdx.x * 2;
+
+  // The channel loaded by that thread.
+  int32_t ci = blockIdx.x * params.cPerBlock + threadIdx.x * CHANNELS_PER_THREAD;
+
+  if (ci >= params.c || threadIdx.x * CHANNELS_PER_THREAD >= params.cPerBlock) {
+    return;
+  }
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:276]: as you have __syncthreads below, early termination will make the thread hang for ever.

File: onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu [hunk 3]
```diff
@@ -439,37 +626,53 @@
   params.c = num_channels;
   params.groups = num_groups;
   params.hw = params.h * params.w;
+
+  // This will allocate as many blocks as possible to partition HW.
```
Comment (yufenglee) [inline onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:630]: why as many blocks as possible? If too many blocks, will atomicAdd be a bottleneck?

Comment (yufenglee) [reply onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:630]: if hw is much larger than 1024, like 320*320, it should be fine.

File: onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu (hunk 2 above)
Comment (yufenglee) [reply onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:276]: double checked. early exit-threads won't block

**Review by yufenglee — APPROVED** [review]
:shipit:

File: onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu (hunk 3 above)
Comment (tianleiwu) [reply onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu:631]: Added some comments

**Review by yufenglee — DISMISSED** [review]
:shipit:

**Review by yufenglee — APPROVED** [review]

**Pull request merged**